### PR TITLE
Drv scpi

### PR DESCRIPTION
**Pull request by luroche**
Working with chan

### CODE REVIEW

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py [hunk 1]
```diff
@@ -17,35 +18,47 @@
...
-    '''
-    def __init__(self, data_type: DrvScipiCmdTypeE, port: str, payload: str|DrvScpiHandlerC):
-        self.data_type: DrvScipiCmdTypeE = data_type
+    "Principal class of the driver. Hold the data to be sent to the device."
+    def __init__(self, data_type: DrvScpiCmdTypeE, port: str,\
+                 payload: str|DrvScpiSerialConfC|None = None):
+        self.data_type: DrvScpiCmdTypeE = data_type
         self.port: str = port
-        self.payload: str|DrvScpiHandlerC = payload
+        self.payload: str|DrvScpiSerialConfC = payload
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: missing status for RESP type. If payload is none no create attribute.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 2]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
         result: list = re.findall(r"-?\d*\.?\d+", data)
         if len(result) < 2:
             msg_decode = float(result[0])
-        elif len(result) == 2:
-            msg_decode = float(result[0]) * 10 ** int(result[1])
         else:
-            raise DrvScpiErrorC(message="Error decoding data", error_code=1)
+            msg_decode = float(result[0]) * 10 ** int(result[1])
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py:80]: Improve for all compatible devices

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 3]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
...
             data (bytes): Value to decode and split.
         Returns:
             msg_decode (List[str]): Message decoded and splited.
         Raises:
-            - None
+            - None.
         """
         data_dec = data.decode("utf-8")
-        msg_decode = data_dec.split(f"{self.__separator}")
+        msg_decode = data_dec.split(' ')
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: use separator for split. Previous version is working properly

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 4]
```diff
@@ -50,28 +39,26 @@ def __str__(self) -> str:
...
+        self.__serial: Serial = Serial(port               = serial_conf.port,
+                                       baudrate           = serial_conf.baudrate,
+                                       bytesize           = serial_conf.bytesize,
+                                       parity             = serial_conf.parity,
+                                       stopbits           = serial_conf.stopbits,
+                                       timeout            = serial_conf.timeout,
+                                       write_timeout      = serial_conf.write_timeout,
+                                       inter_byte_timeout = serial_conf.inter_byte_timeout)
+        self.__separator: str = serial_conf.separator
+        port = self.__serial.port.split('/')[-1]
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: store port chan name as an attribute and process only on initialization

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 5]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
...
             - msg (str): Message to send.
         Returns:
-            - None
+            - None.
         Raises:
-            - None
-        """
+            - None.
+        '''
+        port = self.__serial.port.split('/')[-1]
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: Remove and use __rx_chan_name

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 6]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
...
+            - None.
         Raises:
-            - None
-        """
-        self.send_msg(msg)
-        response: list = self.receive_msg()
-        return response
+            - None.
+        '''
+        port = self.__serial.port.split('/')[-1]
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: use port attribute

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 7]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
...
+        log.info(f"Reading port: {port}...") # pylint: disable=logging-fstring-interpolation
+        msg_read = self.__serial.readline() #TODO: Check if this is the best way to read
+        if len(msg_read) > 0:
+            msg_read_decoded = self.decode_and_split(msg_read)
+            send_data = DrvScpiCmdDataC(port = self.__serial.port, data_type = DrvScpiCmdTypeE.RESP,
+                                        payload = msg_read_decoded)
+            self.__rx_chan.send_data(send_data)
+            self.wait_4_response = False
+        else:
+            log.error(f"Error during send port: {port}") # pylint: disable=logging-fstring-interpolation
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: is not an error. Is an info message. Is expected to raise timeout multiple times until message is received.

Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: If read failed, incrmeent num_attempts

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 8]
```diff
@@ -82,86 +69,69 @@ def decode_numbers(self, data: str) -> float:
...
+            - None.
         Returns:
-            - None
+            - None.
         Raises:
-            - None
-        """
+            - None.
+        '''
         self.__serial.close()
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py:148]: Remove chan when close handler

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 9]
```diff
@@ -16,90 +17,110 @@
...
-    '''
-    def __init__(self, tx_buffer: SysShdChanC, working_flag: Event):
-        self.__used_dev: dict = {}
+    "Returns a removable version of the DRV command."
+    def __init__(self, working_flag: Event, tx_scpi_long: int):
+        self.__used_dev: Dict(str, DrvScpiHandlerC) = {}
         self.working_flag: Event = working_flag
-        self.tx_buffer: SysShdChanC = tx_buffer
+        max_message_size = 300 #TODO: want to know the maximum message size.
+        self.tx_scpi: SysShdIpcChanC = SysShdIpcChanC(name = "cola_SCPI",
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: change name to `tx_scpi`

Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: define `max_message_size` and `max_message_size` as constants at the begin of the file.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 10]
```diff
@@ -16,90 +17,110 @@
...
 #######################              ENUMS               #######################
 
 #######################             CLASSES              #######################
 class DrvScpiNodeC:
-    '''Returns a removable version of the DRV command.
-    '''
-    def __init__(self, tx_buffer: SysShdChanC, working_flag: Event):
-        self.__used_dev: dict = {}
+    "Returns a removable version of the DRV command."
+    def __init__(self, working_flag: Event, tx_scpi_long: int):
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Init function is not the as the architecture defined.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 11]
```diff
@@ -16,90 +17,110 @@
...
 
 #######################          MODULE IMPORTS          #######################
 from .drv_scpi_iface import DrvScpiHandlerC
-from .drv_scpi_cmd import DrvScipiCmdTypeE, DrvScpiCmdDataC
+from .drv_scpi_cmd import DrvScpiCmdDataC, DrvScpiCmdTypeE, DrvScpiSerialConfC
 
 #######################              ENUMS               #######################
 
 #######################             CLASSES              #######################
 class DrvScpiNodeC:
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Add inheritance from SysShdNodeC

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 12]
```diff
@@ -16,90 +17,110 @@
...
-    '''Returns a removable version of the DRV command.
-    '''
-    def __init__(self, tx_buffer: SysShdChanC, working_flag: Event):
-        self.__used_dev: dict = {}
+    "Returns a removable version of the DRV command."
+    def __init__(self, working_flag: Event, tx_scpi_long: int):
+        self.__used_dev: Dict(str, DrvScpiHandlerC) = {}
         self.working_flag: Event = working_flag
-        self.tx_buffer: SysShdChanC = tx_buffer
+        max_message_size = 300 #TODO: want to know the maximum message size.
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Use a constant

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 9 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Both of them must be constants

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 13]
```diff
@@ -16,90 +17,110 @@
...
-        #Add device
-        if cmd.payload is DrvScpiHandlerC:
-            if cmd.data_type is DrvScipiCmdTypeE.ADD_DEV:
-                if self.__used_dev.get(cmd.port) is None:
-                    self.__used_dev[cmd.port] = cmd.payload
+        # Add device
+        if cmd.data_type == DrvScpiCmdTypeE.ADD_DEV:
+            log.info("Adding device...")
+            if cmd.port in self.__used_dev:
+                log.error("Device already exist")
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Change to warning?

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 14]
```diff
@@ -16,90 +17,110 @@
...
+                    log.info("Device added")
                 else:
-                    log.error("The port is already configured")
+                    log.error("The device could not be added")
+
+        # Delete device
+        elif cmd.data_type == DrvScpiCmdTypeE.DEL_DEV:
+            log.info("Deleting device...")
+            if cmd.port in self.__used_dev:
+                del self.__used_dev[cmd.port]
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Before deleteing handler, apply close function

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 15]
```diff
@@ -16,90 +17,110 @@
...
+                else:
+                    log.error("Message not valid")
             else:
-                log.error("Can`t apply command. \
-                        Error in command format, check command type and payload type")
+                log.error("First add device")
 
+        # Error
+        else:
+            log.error("Can`t apply command")
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Change message to `Unknown command: Can't apply command`

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 16]
```diff
@@ -16,90 +17,110 @@
...
-                    command : DrvScpiCmdDataC = self.tx_buffer.receive_data() # type: ignore
-                    self.__apply_command(command)
-            except Exception as err:
-                log.error(f"Error while receiving CAN message\n{err}")
-        self.stop()
-
-
-    def stop(self) -> None:
-        '''TODO: Poner titulo.
+        for __, handler in self.__used_dev.items():
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: use `values()` function

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 17]
```diff
@@ -16,90 +17,110 @@
...
-        log.critical("Stopping SCPI thread.")
-        self.working_flag.clear()
+        try:
+            if not self.tx_scpi.is_empty():
+                # Ignore warning as receive_data return an object,
+                # which in this case must be of type DrvScpiCmdDataC
+                command : DrvScpiCmdDataC = self.tx_scpi.receive_data() # type: ignore
+                log.info(f"Command to apply: {command.data_type.name}") # pylint: disable=logging-fstring-interpolation
+                self.__apply_command(command)
+                self.__receive_response()
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: receive response must be applied always

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 18]
```diff
@@ -16,90 +17,110 @@
...
+        except ValueError as err:
+            log.error(f"Error while applying/removing filter with error {err}") # pylint: disable=logging-fstring-interpolation
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: In which case raise this exception?

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 19]
```diff
@@ -16,90 +17,110 @@
...
+        except Exception: # pylint: disable=broad-except
+            log.error("Error in SCPI thread")
+            self.working_flag.clear()
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: We really should stop it?

File: code/drv_scpi/tests/scpi_node.py [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+#######################            FUNCTIONS             #######################
+if __name__ == '__main__':
+    # Flag to know if the SCPI is working
+    _working_scpi = Event()
+    _working_scpi.set()
+    #Create the thread for SCPI
+    scpi_node = DrvScpiNodeC(working_flag = _working_scpi, tx_scpi_long = 400)
+    try:
+        while 1:
```
Comment (mariuscrsn) [inline code/drv_scpi/tests/scpi_node.py]: When inheritance to SysShdNodeC is add, it must be removed

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py (hunk 1 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Use kwargs

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py [hunk 21]
```diff
@@ -17,35 +18,47 @@
...
-    '''
-    Principal class of the driver.
-    '''
-    def __init__(self, port: str, payload: DrvScpiHandlerC, chan: SysShdChanC):
-        self.port : str= port
-        self.payload : DrvScpiHandlerC = payload
-        self.chan : SysShdChanC = chan
+class DrvScpiSerialConfC:
+    "Configuration of SCPI device."
+    def __init__(self, port: str,           separator: str,
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py:76]: Document function

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py (hunk 4 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: use __rx_chan_name for port attrib

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py [hunk 22]
```diff
@@ -50,28 +39,26 @@ def __str__(self) -> str:
...
+                                       baudrate           = serial_conf.baudrate,
+                                       bytesize           = serial_conf.bytesize,
+                                       parity             = serial_conf.parity,
+                                       stopbits           = serial_conf.stopbits,
+                                       timeout            = serial_conf.timeout,
+                                       write_timeout      = serial_conf.write_timeout,
+                                       inter_byte_timeout = serial_conf.inter_byte_timeout)
+        self.__separator: str = serial_conf.separator
+        port = self.__serial.port.split('/')[-1]
+        self.__rx_chan: SysShdIpcChanC = SysShdIpcChanC(name = f"RX_{port}")
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: rx_{__rx_chan_name}

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py (hunk 7 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py]: Log error message when achieve max_attempts

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 23]
```diff
@@ -16,90 +17,110 @@
...
 
 #######################             CLASSES              #######################
 class DrvScpiNodeC:
-    '''Returns a removable version of the DRV command.
-    '''
-    def __init__(self, tx_buffer: SysShdChanC, working_flag: Event):
-        self.__used_dev: dict = {}
+    "Returns a removable version of the DRV command."
+    def __init__(self, working_flag: Event, tx_scpi_long: int):
+        self.__used_dev: Dict(str, DrvScpiHandlerC) = {}
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py:46]: Add super init

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 18 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Remove it

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 19 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Change to critical

Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Remove clear function

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py (hunk 2 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py:80]: TODO: can't finish yet

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 12 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 13 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 14 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 15 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/SUPPORTED_CMDS.md
Comment (mariuscrsn) [inline code/drv_scpi/SUPPORTED_CMDS.md:1]: Add an comment on readme with a link to this page

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py [hunk 24]
```diff
@@ -17,35 +18,82 @@
...
-    ADD_DEV = 3
+class DrvScpiCmdTypeE(Enum):
+    "Types of commands to be sent to the device."
+    ADD_DEV     = 0
+    DEL_DEV     = 1
+    WRITE       = 2
+    WRITE_READ  = 3
+    RESP        = 4
+
+class DrvScpiStatusE(Enum):
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Remove it, SysShdNodeStatusE will be used instead

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py [hunk 25]
```diff
@@ -17,35 +18,82 @@
...
+            if hasattr(self, 'payload'):
+                if self.data_type == DrvScpiCmdTypeE.ADD_DEV:
+                    if not isinstance(self.payload, DrvScpiSerialConfC):
+                        raise TypeError("No exist payload or must be a DrvScpiSerialConfC object")
+                elif (self.data_type == DrvScpiCmdTypeE.WRITE or \
+                    self.data_type == DrvScpiCmdTypeE.WRITE_READ or \
+                    self.data_type == DrvScpiCmdTypeE.RESP) and (not isinstance(self.payload, str)):
+                    raise TypeError("No exist payload or must be a string")
+            else:
+                log.error("No exist payload")
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: It must raise an exception.

Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Check if exists any exception more suitable than TypeError. Something like ArgError, ParamError....

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py [hunk 26]
```diff
@@ -17,35 +18,82 @@
...
+        self.__dict__.update(kwargs)
+        if self.data_type is not DrvScpiCmdTypeE.DEL_DEV:
+            if hasattr(self, 'payload'):
+                if self.data_type == DrvScpiCmdTypeE.ADD_DEV:
+                    if not isinstance(self.payload, DrvScpiSerialConfC):
+                        raise TypeError("No exist payload or must be a DrvScpiSerialConfC object")
+                elif (self.data_type == DrvScpiCmdTypeE.WRITE or \
+                    self.data_type == DrvScpiCmdTypeE.WRITE_READ or \
+                    self.data_type == DrvScpiCmdTypeE.RESP) and (not isinstance(self.payload, str)):
+                    raise TypeError("No exist payload or must be a string")
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Check if status exists for RESP

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py (hunk 2 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py:80]: Add todo comment on code

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 27]
```diff
@@ -16,90 +18,123 @@
...
-class DrvScpiNodeC:
-    '''Returns a removable version of the DRV command.
-    '''
-    def __init__(self, tx_buffer: SysShdChanC, working_flag: Event):
-        self.__used_dev: dict = {}
-        self.working_flag: Event = working_flag
-        self.tx_buffer: SysShdChanC = tx_buffer
+class DrvScpiNodeC(SysShdNodeC):
+    "Returns a removable version of the DRV command."
+    def __init__(self, name: str, working_flag: Event, cycle_period: int):
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Remove name, is implicit: `can_sniffer`. Always is the same.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 28]
```diff
@@ -16,90 +18,123 @@
...
+        '''
+        Args:
+            - working_flag (Event): Flag to know if the SCPI is working.
+            - cycle_period (int): Period of the cycle.
+        Raises:
+            - None.
+        '''
+        self.__used_dev: Dict(str, DrvScpiHandlerC) = {}
+        self.tx_scpi: SysShdIpcChanC = SysShdIpcChanC(name = 'tx_scpi',
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Move name as an constant at the beginning of the file.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 29]
```diff
@@ -16,90 +18,123 @@
...
+        Args:
+            - None.
+        Returns:
+            - None.
+        Raises:
+            - None.
+        '''
+        if not self.tx_scpi.is_empty():
+            command : DrvScpiCmdDataC = self.tx_scpi.receive_data() # type: ignore
+            print(command.__dict__)
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Remove print

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py [hunk 30]
```diff
@@ -16,90 +18,123 @@
...
-            - None
+            - None.
         Raises:
-            - None
+            - None.
         '''
-        log.critical("Stopping SCPI thread.")
-        self.working_flag.clear()
+        log.info("Stopping SCPI node...")
+        # for device in self.__used_dev.values():
```
Comment (mariuscrsn) [inline code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Uncomment it

**Review by mariuscrsn — CHANGES_REQUESTED** [review]

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py (hunk 2 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_iface.py:80]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py (hunk 24 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Done.

File: code/drv_scpi/SUPPORTED_CMDS.md
Comment (luroche) [reply code/drv_scpi/SUPPORTED_CMDS.md:1]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py (hunk 26 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py:65]: I change it as ValueError

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 27 above)
Comment (mariuscrsn) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: scpi_sniffer

Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 28 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 30 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py (hunk 29 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_node.py]: Done.

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py (hunk 25 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: I change it as ValueError

File: code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py (hunk 26 above)
Comment (luroche) [reply code/drv_scpi/src/scpi_sniffer/drv_scpi_cmd.py]: Done.

File: code/drv_scpi/README.md [hunk 31]
```diff
@@ -2,4 +2,6 @@
 This module implements a node that facilitates bidirectional command communication
 with multiple devices using the SCPI protocol. It enables seamless sending and 
 receiving of commands, allowing efficient control and data exchange with 
-various instruments.
\ No newline at end of file
+various instruments.
+
+View device commands in SUPPORTED_CMDS.md
```
Comment (mariuscrsn) [inline code/drv_scpi/README.md]: Add ref to file

**Review by mariuscrsn — APPROVED** [review]

File: code/drv_scpi/README.md (hunk 31 above)
Comment (luroche) [reply code/drv_scpi/README.md]: Done.

**Review by mariuscrsn — APPROVED** [review]

**Pull request merged**